src/libraries/System.Text.Json/tests/Common/JsonSchemaExporterTests.TestTypes.cs

```diff
@@ -0,0 +1,1320 @@
+                        }
+
+                        DefaultValueAttribute? defaultValueAttr =
+                            GetCustomAttribute<DefaultValueAttribute>(ctx.PropertyInfo?.AttributeProvider) ??
+                            GetCustomAttribute<DefaultValueAttribute>(ctx.PropertyInfo?.AssociatedParameter?.AttributeProvider);
+
+                        if (defaultValueAttr != null)
+                        {
+                            schema["default"] = JsonSerializer.SerializeToNode(defaultValueAttr.Value, ctx.TypeInfo);
+                        }
+
+                        RegularExpressionAttribute? regexAttr =
+                            GetCustomAttribute<RegularExpressionAttribute>(ctx.PropertyInfo?.AttributeProvider) ??
+                            GetCustomAttribute<RegularExpressionAttribute>(ctx.PropertyInfo?.AssociatedParameter?.AttributeProvider);
+
+                        if (regexAttr != null)
+                        {
+                            schema["pattern"] = regexAttr.Pattern;
+                        }
+                    }
+                });
+
+            yield return new TestData<ClassWithJsonPointerEscapablePropertyNames>(
+                Value: new ClassWithJsonPointerEscapablePropertyNames { Value = new() },
+                ExpectedJsonSchema: """
+                {
+                    "type": ["object","null"],
+                    "properties": {
+                        "~/path/to/value": {
+                            "type": "object",
+                            "properties": {
+                                "Value" : {"type":"integer"},
+                                "Next": {
+                                    "type": ["object","null"],
+                                    "properties": {
+                                        "Value" : {"type":"integer"},
+                                        "Next": {"$ref":"#/properties/~0~1path~1to~1value/properties/Next"}
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                """);
+
+            // Collection types
+            yield return new TestData<int[]>([1, 2, 3], ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":"integer"}}""");
+            yield return new TestData<List<bool>>([false, true, false], ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":"boolean"}}""");
+            yield return new TestData<HashSet<string>>(["one", "two", "three"], ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":["string","null"]}}""");
+            yield return new TestData<Queue<double>>(new([1.1, 2.2, 3.3]), ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":"number"}}""");
+            yield return new TestData<Stack<char>>(new(['x', '2', '+']), ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":"string"}}""");
```

Comment:
Might consider adding a `minLength: 1, maxLength: 1` to the `items` subschema for `char` support.

Reply:
Good idea, although `maxLength` would need to be set to 2 to account for surrogate pairs. (Assuming that `maxLength` and `minLength` check the number of UTF code points and not something different.)

Reply:
JSON Schema string length is calculated as Unicode code point, not chars/bytes, so `maxLength` would still be 1.

Refs: https://json-schema.org/draft/2020-12/json-schema-validation#section-6.3.1 and https://www.rfc-editor.org/rfc/rfc8259.html#section-8.2

We also have a test for this:

https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/9fc880bfb6d8ccd093bc82431f17d13681ffae8e/tests/draft2020-12/maxLength.json#L29-L33

Reply:
For whatever reason I was thinking that surrogate pairs can fit in one char, however that is clearly not the case. You're correct that maxLength should be 1.